app/src/main/java/com/example/capstone/activity/HomescreenActivity.java

```diff
@@ -102,6 +143,31 @@ public void onMapClick(LatLng point) {
                 selectedLatLng = point;
             }
         });
+
+        map.setOnCameraMoveListener(new GoogleMap.OnCameraMoveListener(){
+            @Override
+            public void onCameraMove() {
+                CameraPosition cameraPosition = googleMap.getCameraPosition();
+                if (currentZoom == cameraPosition.zoom) {
+                    return;
+                }
+                if (mapGestureHandler == null) {
+                    return;
+                }
+
+                currentZoom = cameraPosition.zoom;
+
+                if (currentZoom < MIN_ZOOM) {
+                    mapGestureHandler.onZoomChange(MIN_ZOOM, currentLatLng);
+                }
+                if (currentZoom > MAX_ZOOM) {
+                    mapGestureHandler.onZoomChange(MAX_ZOOM, currentLatLng);
+                }
+
+                mapGestureHandler.onZoomChange((int) currentZoom, currentLatLng);
```

Comment:
This is a bug. This line will be run everytime even if currentZoom < MIN_ZOOM or currentZoom > MAX_ZOOM. Should be in if-else block